pkg/datastructures/hashtable/main.go

```diff
@@ -0,0 +1,55 @@
+package hashtable
+
+type (
+	item struct {
+		next  *item
+		key   string
+		value string
+	}
+	bucket struct {
+		node   *item
+		length int
+	}
+)
+
+var table = make([]*bucket, 100)
+
+// hash function, adds up the rune values of the characters in the key
+func hash(key string) int {
+	sum := 0
+	for _, v := range key {
+		sum += int(v)
+	}
+	sum %= 100
```

Comment:
Consider using a Prime Number rather than `100`

Reply:
Is this because `x % 100` gives the same value for 140 and 40?

Reply:
Not exactly, A subtle way to put it would be **This is how maths works** 😁

Basically modulo by primes give more distributed hashes, something to do with **co-primes** a thorough ans is given 
[here](https://stackoverflow.com/questions/1145217/why-should-hash-functions-use-a-prime-number-modulus)

Reply:
Thank you, makes sense.